pkg/client/leaderelectionconfig/config.go

```diff
@@ -43,4 +43,10 @@ func BindFlags(l *componentbaseconfig.LeaderElectionConfiguration, fs *pflag.Fla
 	fs.StringVar(&l.ResourceLock, "leader-elect-resource-lock", l.ResourceLock, ""+
 		"The type of resource object that is used for locking during "+
 		"leader election. Supported options are `endpoints` (default) and `configmaps`.")
+	fs.StringVar(&l.ResourceName, "leader-elect-resource-name", l.ResourceName, ""+
```

Comment:
I am pretty nervous about offering these options.
1) it's now fairly easy to misconfigure and end up with multiple active controllers, which could be catastrophic (imagine thrashing over load balancers, for example)
2) is it possible to change these flags one replica at a time in a HA configuration? How?

Reply:
> 1. it's now fairly easy to misconfigure and end up with multiple active controllers, which could be catastrophic (imagine thrashing over load balancers, for example)

In most case, we needn't configure these options. We will configure them when we want to seperate some controllers out of kcm.
> 2. is it possible to change these flags one replica at a time in a HA configuration? How?

I don't understand why we want to change these flags for a inflight controller.
If we want to do it, firstly stop the standby kcms, change all kcm's flags, restart the active kcm, restart the standby kcms.

Reply:
The scheduler provides these flags before, it's designed for multi-scheduler, clients will only configure these flags for the new scheduler, but not to the inflight scheduler.
And now we just let kcm has the same ability.

Reply:
Might be worth looking at https://github.com/kubernetes/enhancements/pull/979. Stopping standby KCMs can be problematic as component status (CS) is always the local KCM. So it will make the KAS appear unhealthy via CS requests. Depending on the users/CPs monitoring system that may cause problems. More importantly the set of safe mechanisms for migrating from 1 CM to multiple CMs is not documented anywhere.

Reply:
> Might be worth looking at [kubernetes/enhancements#979](https://github.com/kubernetes/enhancements/pull/979). Stopping standby KCMs can be problematic as component status (CS) is always the local KCM. So it will make the KAS appear unhealthy via CS requests. Depending on the users/CPs monitoring system that may cause problems. 

I think we'd better not to change leader election object in arbitrary for a inflight component. And I think these options are required for migrating multiple CMs.
> More importantly the set of safe mechanisms for migrating from 1 CM to multiple CMs is not documented anywhere.

I am glad to add a document about it in the next PR, and where should I put the document?

Reply:
If we switch the resourceLock from endpoints to configmaps for a inflight CM, I think it is the same situation. It should be a new topic for migration lock of inflight CM. This PR doesn't add risk for clients.